self-signed-chain

qinohe opened the pull request:
self-signed-chain added

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+
+   * Install them from unknown parties at all.
+
+A chain will need at least the next ingredients:
```
incorrect - intermediate CAs are not needed. A top level CA can directly sign certificates. This intermediates are often used to lock away the root CA keys to not replace anything in case of a compromise.

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+.. Tip::
+    When creating the server certificate make sure the **CN - common name**
+    is in fact the the **FQDN - fully qualified domain name**.
+    You can find it on **Linux/Unix** with this command ``hostname -f``
```
IMHO this needs a column (":")

qinohe replied:
Yup done

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+The Nextcloud Authority
+-----------------------
+
+Go to **trust/Authorities** create a new CA for Nextcloud and save it.
```
please recheck - menu elements usually start upcase

qinohe replied:
Forgive my ignorance but which item do you mean **trust/Authorities**?
Or that 'The Nextcloud Authority' should be downcase??

fabianfrz replied:
![screenshot_20180711_221315](https://user-images.githubusercontent.com/3713914/42597000-a3bbef62-8557-11e8-939b-5de1dd260b16.png)

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+OPNsense needs to be made aware of the Nextcloud chain we are creating.
+
+   * Download the **CA.crt** and upload it back to OPNsense in a secure way.
+   * For this you can use ``scp`` see ``man scp``
```
the "see" part should be in round brackets because it does not belong to the sentence.

qinohe replied:
Right, I see ;)

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+
+Download the intermediate CA and install it to your browser
+
+   * Head to the webgui **trust/Authorities** export **nextcloud-intermediate-ca**
```
same here

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+Download the intermediate CA and install it to your browser
+
+   * Head to the webgui **trust/Authorities** export **nextcloud-intermediate-ca**
+   * Back to the browser, open **preferences/certificate/authorities**
```
and here

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+====================== =================================== ========================================
+ **Descriptive name**   cloudserver-cert                    *Choose a name that makes sense to you*
+ **Method**             create a server certificate         *Main purpose of certificate*
+ **Common Name**        cloud.localdomain                   *Should reflect the FQDN*
```
nowadays you should use SAN (subject alternative name) and use CN as a fallback for old SW.

qinohe replied:
Will change it, though I was basing myself on the forms presented in OPNsense
I will sleep over that one and how I will translate that to the form.

fabianfrz commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,271 @@
+-  * **/etc/ssl/localcerts** will be alright or choose your own prefered location.
+   * Edit the webserver config to use the certificate, the key is included in the ***.crt**
+   * There are other ways to do this.
+   * Sane permissions, '400' read only owner is sufficent.
```
I prefer adding server keys to the corresponding server into its directory or a subdirectory if there are multiple key/cert pairs.

For example `/etc/nginx/nextcloud.key` and `/etc/nginx/nextcloud.crt`

I think such setups are more common.

qinohe replied:
Okay, I will add this as a way to extract the archive too

fabianfrz commented:
@qinohe I have added some comments and there are currently two issues: your git client seems to be incorrectly configured so github cannot show you as the author of the files which has the issue that your copyright is not documented. The other thing - I don't know if it is an issue but I would suggest to switch to the default theme when creating the screenshots (only for the screenshots - you can switch back afterwards). @fichtner @AdSchellevis should know how the docs should look like. Some people may print them and you usually cannot read that in case of a laser printer.

qinohe commented:
Thank you fabian so much for going through the document, I will need some time to investigate for I do not even completely understand some of your comments regarding the markup, will get back if I can't solve it.
About git, I have no clue yet, will investigate.
About the theme, I haven't thought about that for a second will change it.

fabianfrz commented:
if you don't do a commit, it is easy to change the old one using `--amend --author "First Last <Mail>"`

fabianfrz commented:
note: amend requires a forced push (`git push -f`)

qinohe commented:
Haha, I used the shell to upload to github, but I used the browser to push it to you guys.
It did it like that to not f**k up everything, this is my first commit in years, and the remote commit ever ;)
Franco already told "git can be hell" so that's why I did it like this, I will dive into git and try to master these basics..

qinohe commented:
Since I'm here I have a question too, can I add a custom 'color.css'  for RGB colors and add it to conf.py?
I was already noted to not to batch  commits I already had it in place but stripped it out.
If you don't allow it I won't even commit it, but it's a nice addition to color text.
Don't worry I will only add it once this is done, thanks mark

qinohe commented:
Right I'm stuck and not able to amend myself as the author
I have no clue how to change it, after this is done I will first create 'git config', so sad I forgot that. :(
Also next time I do a commit that will be the 'pristine' directory and work in a copy of that one, that's how we learn I guess.
Anyway, is there still a way to change it, I already tried a clone but it don't work like that.
I keep getting a message:  `git commit --amend --author is not 'Name <email>' and matches no existing author`

fichtner commented:
@qinohe thanks so far, I'll jump into the review soon. for now...

```
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly:

    git config --global user.name "Your Name"
    git config --global user.email you@example.com

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author
```

Fabian is right about the theme for the screenshot. Light theme is way better for printing. I hope that's not too much to ask, it's a bit of work to do this in the first place.

qinohe commented:
Thanks fichtner you're welcome. It's no problem at all, I did create them right away yesterday..;)
I just now did the git config and 'push -f' and the strange thing is the images are now 'mine' but the RST document is not.
Do it again?

fabianfrz commented:
if @fichtner does a squashed commit, github will fix it.

fichtner commented:
Now it's all good with the author info, but the pictures are the old ones?  Not to worry, can still squash multiple commits on this branch afterwards.

qinohe commented:
Yes ,I know but since I don't know too much about versioning I waited till this worked out okay, and it did and will be fixed with your squashed commit, I get it.
Now I will fix based on fabians comments and swap the pictures, haha, I'm moving the speed of snail, but I will get there, I promise..;), it's worse I keep you guys busy too

qinohe commented:
Hi guys, have done some enhancements pics changed hope you like it ;) greetz mark

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+==================================================
+Setup Self Signed Certificate Chains with OPNsense
```
can this be consolidated to "self-signed" everywhere?

qinohe replied:
I you wish, yes no problem.

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+nothing wrong with that certificate if we use a real world CA, but we don't. We
+create our own chain so that one has no purpose once done.
+
+Should you even consider using **self signed certificate chains**? in this age of free available 
```
stray "?"

qinohe replied:
around , yup ;) removed

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+certificates?
+
+   * Self signed certificate are just as secure as real world certificates.
+   * They are trustworthy chains, you **know** all parties
```
Bullet points are sentences except here...

qinohe replied:
Elaborate a little please, why aren't they sentences?

fichtner replied:
missing "." :)

qinohe replied:
Ah tsss.. ;) Yes I walk trough all sentences once more for forgotten or wrong punctuation marks. Thanks

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+   * Self signed certificate are just as secure as real world certificates.
+   * They are trustworthy chains, you **know** all parties
+   * Intranets are often set up with these chains
```
and here

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+more than that, a single certificate.
+
+Look at the default install, one certificate is created for the webgui/dashboard. There is
+nothing wrong with that certificate if we use a real world CA, but we don't. We
```
In technical documentation contractions are best avoided. But I leave this up to you in this how-to. It's made for humans after all 👍

qinohe replied:
Yeah well I agree and I have stripped a few already, I'll rip the 'forgotten' if it stays readable ;)

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+What you shouldn't do with a self signed chain
+
+   * Use them for a web facing server
```
web-facing

fabianfrz replied:
I am not sure about it - dashes are common in german but not in english

fichtner replied:
it's ok http://www.whitesmoke.com/using-hyphens

the word is "webfacing" which looks silly so it's "web-facing"

qinohe replied:
Hmm, I actually looked that on up when I needed it in Google translate Dutch;
web facing --> 'naar het web gericht'
web-facing --> 'web-facing"  
It don't matter to me and I guess anyone reading both gets the picture ;)
Go with dash then?

qinohe replied:
;) and in German there is a result
web-facing --> web-zugewandt
I think Google is not yet up to speed with Dutch ..;)

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+What you shouldn't do with a self signed chain
+
+   * Use them for a web facing server
+   * Install them unless you are the issuer
```
without checking fingerprints... you may have a communication party you want to reach for one reason or another

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+What you shouldn't do with single self signed certs
+
+   * Install them from unknown parties at all.
```
If you have the chain, there is no need to add the individual certificates as the chain would already be (un)trusted, yep. Maybe this could be made a little clearer?

qinohe replied:
Right you should, then there would be no need to do so, it's to cryptic I see now,  I could have better said 'Install them at all' , thanks good sleuth ;)

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+   * Install them from unknown parties at all.
+
+A chain will need at least a CA and certificate, an intermediate CA is not needed, but in case of a 
```
first comma is more of a semicolon ;)

qinohe replied:
Grammatical I was never very strong ;)

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+.. Note::
+
+    This ducument uses **CN - Common Name** should be read as: **SAN - Subject Alternative Name** and 
```
document

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+    This ducument uses **CN - Common Name** should be read as: **SAN - Subject Alternative Name** and 
+    will be used if present.
+
+Please backup before you procede.
```
proceed

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+-------------
+
+The first certificate to create is the **CA**. The only thing this CA
+does is sign the **intermediate CA** next in the line of trust.
```
Self-signed root CAs anchor trust chains. It's important to stress this. OpenSSL requires it, it mostly doesn't care about intermediates, especially in S/MIME signatures. But anyway... just a comment.

qinohe replied:
Will add a note on this..

fichtner commented on source/manual/how-tos/self-signed-chain.rst:
```diff
@@ -0,0 +1,339 @@
+
+::
+
+   openssl pkcs12 -in nextcloud-crt.p12 -nodes -out nextcloud.key -nocerts
```
nice to post these, people tend to have to look them up more often than not :)

qinohe replied:
you're welcome, it also removes/lowers the change of people looking it up on some crappy site and asking on the forum why the guide doesn't work .. and it does ;)

qinohe replied:
It was on request btw.. from fabian :)

qinohe commented:
Hi guys, stripped all :': you may say if it's all still clear..
Went through all text and punctuation marks and think I fixed most, getting a little blind for the same text ;)
Added a note on 'trust anchor' but left S/MIME out for now, there's no entry for that, yet  ;)

fichtner commented:
Nice work @qinohe ! We'll wrap this up but you can always submit corrections later... so... merged, thanks! :)

qinohe commented:
Thank you @fitchner :) My way to say: Thank you for guys OPNsense!

fichtner commented:
Well, it's only as good as its contributors. :)

qinohe commented:
Hey guys, updated this page so it's up to date with 19.7.1 
I decided to not remove but to give notes because there are users using older versions.
When it becomes obsolete we can remove the references, at least that seems the best way to me, correct me if I'm wrong ;)

fichtner commented:
Wouldn't worry so much about colours, but maybe @AdSchellevis has an opinion.

AdSchellevis commented:
@qinohe I'm not sure what you want to add, if you create a separate PR with the css change you want to make, I'll take a look.

qinohe commented:
Hi Franco,
You must have seen I was trying to edit this oldie...;)

> @qinohe I'm not sure what you want to add, if you create a separate PR with the css change you want to make, I'll take a look.

Hi Ad,
I think the question is not relevant anymore since I now use standard markups, like Note or Warning.
Basically  I was coloring sentences, which looks nice, but deviates from your default theme, so I wouldn't do that if I were you.
Anyway, thanks for getting back on this one.

Regards, mark
